ebu_tt_live/node/denester.py

```diff
@@ -89,9 +90,20 @@ def denest(document):
             unnested_divs.extend(DenesterNode.recurse(div, dataset))
         unnested_divs = DenesterNode.combine_divs(unnested_divs)
         unnested_divs = DenesterNode.check_p_regions(unnested_divs)
-        #TODO: do we have to remove the old divs from the body's ordered content? body can contain metadata or divs
         document.binding.body.div.clear()
-        document.binding.body.div.extend(unnested_divs)
+        # Remove the old divs from the body's ordered content, before
+        # adding the new divs. Note that, as is, this will remove only
+        # div elements from the body, and append new divs, so if there
+        # are foreign namespace elements interleaved with the div
+        # children, the interleaving will break.
+        # A better pattern for this is in `recurse_span`, but there
+        # wasn't enough time to implement that here.
+        for e in reversed(document.binding.body.orderedContent()):
```

Comment:
Why do you need to put this in `reversed`? I tried removing it and all the tests still passed. Is there a case where something would go wrong if you didn't call reversed?

Reply:
Because I might remove the thing, and we learned that if you remove a thing while iterating forward through the list then the iterator doesn't know you removed it and you miss an item.